### PR TITLE
Take latest changes from common, migration of shared prefs

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**
Adapt to: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1465

### CODE REVIEW

File: userappwithbroker/build.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@ allprojects {
     }
 }
 
-def adalVersion = "3.1.2"
+def adalVersion = "3.1.3"
```
Comment (rpdome) [inline userappwithbroker/build.gradle:19]: intentional?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-bin.zip
```
Comment (rpdome) [inline gradle/wrapper/gradle-wrapper.properties:6]: nit: intentional?

File: adal/build.gradle [hunk 3]
```diff
@@ -183,7 +184,7 @@ task sourcesJar(type: Jar) {
 task javadoc(type: Javadoc) {
     failOnError false
     source = android.sourceSets.main.java.srcDirs
-    classpath += configurations.compile
+    classpath += configurations.api
```
Comment (rpdome) [inline adal/build.gradle:187]: What difference does this make?

**Review by rpdome — APPROVED** [review]

File: adal/build.gradle (hunk 3 above)
Comment (AdamBJohnsonx) [reply adal/build.gradle:187]: It exists when we move to gradle 7.0

File: userappwithbroker/build.gradle (hunk 1 above)
Comment (AdamBJohnsonx) [reply userappwithbroker/build.gradle:19]: Yes.  We moved ADAL versions, should be up to date.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (AdamBJohnsonx) [reply gradle/wrapper/gradle-wrapper.properties:6]: I tried to standardize these wrapper declarations a little.

File: adal/build.gradle (hunk 3 above)
Comment (shahzaibj) [reply adal/build.gradle:187]: Is the `compile` no longer available?

File: adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java [hunk 4]
```diff
@@ -629,12 +635,12 @@ public void saveAccount(String accountName) {
             return;
         }
 
-        SharedPreferencesFileManager prefs = new SharedPreferencesFileManager(mContext, KEY_SHARED_PREF_ACCOUNT_LIST);
-        String accountList = prefs.getString(KEY_APP_ACCOUNTS_FOR_TOKEN_REMOVAL);
+        INameValueStorage<String> prefs = mComponents.getNameValueStore(KEY_SHARED_PREF_ACCOUNT_LIST, String.class);
```
Comment (shahzaibj) [inline adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java]:
```suggestion
        final INameValueStorage<String> prefs = mComponents.getNameValueStore(KEY_SHARED_PREF_ACCOUNT_LIST, String.class);
```

File: adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java [hunk 5]
```diff
@@ -629,12 +635,12 @@ public void saveAccount(String accountName) {
...
+        String accountList = prefs.get(KEY_APP_ACCOUNTS_FOR_TOKEN_REMOVAL);
```
Comment (shahzaibj) [inline adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java:639]:
```suggestion
        final String accountList = prefs.get(KEY_APP_ACCOUNTS_FOR_TOKEN_REMOVAL);
```

**Review by shahzaibj — APPROVED** [review]
LGTM

File: adal/build.gradle (hunk 3 above)
Comment (AdamBJohnsonx) [reply adal/build.gradle:187]: nope

**Pull request merged**